src/main/flight/mixer.c

```diff
@@ -735,18 +750,8 @@ void mixTable(timeUs_t currentTimeUs, uint8_t vbatPidCompensation)
     const float vbatCompensationFactor = vbatPidCompensation ? calculateVbatPidCompensation() : 1.0f;
 
     // Apply the throttle_limit_percent to scale or limit the throttle based on throttle_limit_type
-    if ((currentControlRateProfile->throttle_limit_percent < 100) && (currentControlRateProfile->throttle_limit_type != THROTTLE_LIMIT_TYPE_OFF)) {
-        const float throttleLimitFactor = currentControlRateProfile->throttle_limit_percent / 100.0f;
-        switch (currentControlRateProfile->throttle_limit_type) {
-            case THROTTLE_LIMIT_TYPE_SCALE:
-                throttle = throttle * throttleLimitFactor;
-                break;
-            case THROTTLE_LIMIT_TYPE_CLIP:
-                if (throttle > throttleLimitFactor) {
-                    throttle = throttleLimitFactor;
-                }
-                break;
-        }
+    if (currentControlRateProfile->throttle_limit_type) {
```

Comment:
This should probably check for `currentControlRateProfile->throttle_limit_type != THROTTLE_LIMIT_TYPE_OFF` to make it resistant to the enum being changed.

Reply:
There are some discussions about adding at least one other `throttle_limit_type` scheme.

Reply:
@mikeller 
Usually the enums should be added instead of modified

But ok let's keep it same as it is refactoring anyway

Reply:
@borisbstyle: Totally agreed that they should. On the other hand, the compiler will optimise both versions to the same code, and the more explicit version has more chance of throwing an error if somebody was to change the `enum` in the wrong way, making it more robust.